### PR TITLE
Chore: restore server guide from old restify.com

### PR DESCRIPTION
**Pull request by DonutEspresso**
This restores the server guide from the old restify.com. Also addresses #1459

### CODE REVIEW

File: docs/api/server.md [hunk 1]
```diff
@@ -41,25 +43,58 @@ A restify server has the following properties on it:
...
+// a static route
+server.get('/foo', function(req, res, next) {});
+// a parameterized route
+server.get('/foo/:bar', function(req, res, next) {});
+// a regular expression
+server.get(/^\/([a-zA-Z0-9_\.~-]+)\/(.*)/, function(req, res, next) {});
+// an options object
+server.get({
+    path: '/foo',
+    version: ['1.0.0', '2.0.0]
```
Comment (retrohacker) [inline docs/api/server.md]: Hmm something weird with the markdown here?

File: docs/api/server.md [hunk 2]
```diff
@@ -41,25 +43,58 @@ A restify server has the following properties on it:
...
+```
+
+### address()
 
 Wraps node's [address()](http://nodejs.org/docs/latest/api/net.html#net_server_address).
 
-## listen(port, [host], [callback])
+### listen(port, [host], [callback])
 
 Wraps node's [listen()](http://nodejs.org/docs/latest/api/net.html#net_server_listen_path_callback).
```
Comment (retrohacker) [inline docs/api/server.md]: It doesn't look like GitHub's markdown likes `'`?

File: docs/api/server.md (hunk 1 above)
Comment (sean3z) [inline docs/api/server.md]: Missing a `'` after `2.0.0`
```
['1.0.0', '2.0.0']
```

File: docs/api/server.md (hunk 2 above)
Comment (sean3z) [inline docs/api/server.md]: Correcting the missing `'` above will fix this 😃

Comment (DonutEspresso) [reply docs/api/server.md]: Thanks @sean3z!

**Comment by DonutEspresso** [comment]
BTW the intention for this is to get something back up that at least contains the information - we can work on the copy iteratively.

**Pull request merged**